ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/ScreenConditionState.kt

```diff
@@ -0,0 +1,110 @@
+package com.revenuecat.purchases.ui.revenuecatui.components
+
+import androidx.compose.material3.adaptive.currentWindowAdaptiveInfo
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.onSizeChanged
+import androidx.compose.ui.unit.Density
+import androidx.window.core.layout.WindowWidthSizeClass
+import com.revenuecat.purchases.UiConfig
+import kotlin.math.min
+
+internal class ScreenConditionState(
+    screenSizes: List<UiConfig.AppConfig.ScreenSize>?,
+) {
+    private var breakpoints: List<UiConfig.AppConfig.ScreenSize> = resolveBreakpoints(screenSizes)
+    private var layoutWidthDp: Float? = null
+    private var layoutHeightDp: Float? = null
+    private var windowWidthSizeClass: WindowWidthSizeClass = WindowWidthSizeClass.COMPACT
+
+    var snapshot by mutableStateOf(ScreenConditionSnapshot())
+        private set
+
+    fun updateLayoutSize(widthDp: Float, heightDp: Float) {
```

Comment:
Hmm I'm not in love with making all these methods available to callers, since they could potentially be called somewhere else, when they are only intended to be called by the `rememberScreenConditionState` and `trackScreenCondition` below... But I'm not sure what would be the best approach... 

We could make `rememberScreenConditionState` a companion object function, and would allow us to make some of these private, but wouldn't work for `trackScreenCondition` since that's an extension function of `Modifier`... So I think it's ok to leave as is for now...